plugins/image-source/obs-slideshow.c

```diff
@@ -44,31 +63,56 @@ struct image_file_data {
 	obs_source_t *source;
 };
 
+enum behavior {
+	BEHAVIOR_STOP_RESTART,
+	BEHAVIOR_PAUSE_UNPAUSE,
+	BEHAVIOR_ALWAYS_PLAY,
+};
+
 struct slideshow {
 	obs_source_t *source;
 
 	bool randomize;
+	bool loop;
+	bool restart_on_activate;
+	bool pause_on_deactivate;
+	bool restart;
+	bool manual;
+	bool hide;
+	bool use_cut;
+	bool paused;
+	bool stop;
 	float slide_time;
 	uint32_t tr_speed;
 	const char *tr_name;
 	obs_source_t *transition;
 
 	float elapsed;
-	size_t cur_item;
+	int cur_item;
 
```

Comment:
What's the reason behind this change? Seems to introduce a lot of unnecessary casting later on.